### DISCUSSION TITLE
How to get the output variables ?

### DISCUSSION THREAD

**Discussion by ap766** [OpenModelica/OpenModelica#11418] [Q&A]
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/9495f8e7-2d99-47f3-8817-7a1de468971c)

Here, I have some input variables here

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/2faee96e-9aeb-4672-9800-a68dae1bb6f5)

And here are the output variables.

So, where do the calculations actually happen? is there any way i can get where the calculations values are stored .Actually i am not doing it directly through Openmodelica.I am making a GUI, taking the input variables as input and need to modify the init XML file  such that the output variables are calculated based on input.How can i do that?

I have figured a way that i can modify the XML file such that it reflects in the variables browser / output variables .But however how can i get the calculated values ?(suppose i am not doing it directly in openmodelica)

**Comment by ap766** [comment]
I'm on a tight schedule with this, so a response would be highly valued. 😓

**Comment by adeas31** [comment]
> So, where do the calculations actually happen? is there any way i can get where the calculations values are stored .Actually i am not doing it directly through Openmodelica.I am making a GUI, taking the input variables as input and need to modify the init XML file such that the output variables are calculated based on input.How can i do that?

The values are stored in the mat result file in form of time series.

**Comment by casella** [comment]
If needed, you can read the .mat file using the [DyMat](https://pypi.org/project/DyMat/) python package.

**Comment by lucas-mior** [comment]
You can also configure openmodelica to use other output formats like csv. 

> So, where do the calculations actually happen?

However if you mean intermediate values between steps I don't think that is is possible.

**Comment by casella** [comment]
@ap766 it would help a lot if you briefly explain what are your requirements 😃
